### PR TITLE
Parquet list reading fix

### PR DESCRIPTION
**Pull request by maxulan**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):

- Fixed reading of the files in `Parquet` format containing columns of type `list`. 

...

Detailed description (optional):
This fix is for the case when an input parquet file contains a column of type list which should be skipped. 
Conversion of `Parquet` `lists` into CH arrays is NOT implemented.
This issue was caused by lack of `list` type support in the used version of `arrow` lib.

`arrow` version is updated to latest release: `apache-arrow-0.15.1`.
...

### CODE REVIEW

**Review by alexey-milovidov — COMMENTED** [review]
We should add flatbuffers to contrib. Downloading source code at build time is not an option.

**Comment by maxulan** [comment]
Flatbuffers added to contrib. 
@alexey-milovidov one [integration test (release)](https://clickhouse-test-reports.s3.yandex.net/8334/dbaf47054151dfb8a39f2f00e35c1bbcb4354f70/integration_tests_(release).html) above fails. But it looks like this failure has nothing to do with my PR: `test_parts_delete_zookeeper/test.py::test_merge_doesnt_work_without_zookeeper`.

**Comment by maxulan** [comment]
@alexey-milovidov Could you please review?

**Comment by alesapin** [comment]
conflicts in 19.17.

**Comment by filimonov** [comment]
Opened an FR in apache arrow issue tracker: https://issues.apache.org/jira/browse/ARROW-7845

File: .gitmodules [hunk 1]
```diff
@@ -128,6 +128,9 @@
 [submodule "contrib/icu"]
 	path = contrib/icu
 	url = https://github.com/unicode-org/icu.git
+[submodule "contrib/flatbuffers"]
+	path = contrib/flatbuffers
+	url = https://github.com/google/flatbuffers.git
```
Comment (wesm) [inline .gitmodules]: Note that Flatbuffers is no longer a build dependency of Apache Arrow in C++ (we have vendored the necessary files)

**Comment by wesm** [comment]
Per comments on the Arrow JIRA, if someone can clarify what you mean by "This issue was caused by lack of list type support in the used version of arrow lib." We do support reading ListArray so long as the child type is either a list or a primitive type (but not struct). The main JIRA for this is 

https://issues.apache.org/jira/browse/ARROW-1644

It's being worked on right now, having your feedback on this would be helpful

**Pull request merged**